### PR TITLE
Android popup

### PR DESCRIPTION
**Pull request by thevirtuoso1973**

### CODE REVIEW

File: app/src/main/java/com/jetbrains/handson/mpp/mobile/TrainDataAdapter.kt [hunk 1]
```diff
@@ -29,6 +26,8 @@ class TrainDataAdapter(private val dataSet: List<TrainTimes.Journey>) :
             /*
             Not as clean as it could be due to the API level 23.
              */
+            journey = j
```
Comment (abeshi-softwire) [inline app/src/main/java/com/jetbrains/handson/mpp/mobile/TrainDataAdapter.kt:30]: Do we need to merge `common-format` first?

**Review by abeshi-softwire — COMMENTED** [review]
Do we need to merge common-format first?

File: app/src/main/java/com/jetbrains/handson/mpp/mobile/TrainDataAdapter.kt [hunk 2]
```diff
@@ -78,10 +78,26 @@ class TrainDataAdapter(private val dataSet: List<TrainTimes.Journey>) :
...
+                intent.putExtra("stationChangeNames", holder.journey?.changes?.map {
+                    it.name
+                }?.toTypedArray())
+                val diff = holder.journey?.arrivalTime!! - holder.journey?.departureTime!!
+                val hours = TimeUnit.MILLISECONDS.toHours(diff)
+                val min = TimeUnit.MILLISECONDS.toMinutes(diff) - hours*60
+                intent.putExtra("duration", if (hours == 0L)
+                    "$min minutes"
+                else
+                    "$hours hours, $min minutes")
```
Comment (abeshi-softwire) [inline app/src/main/java/com/jetbrains/handson/mpp/mobile/TrainDataAdapter.kt:98]: Do we need a `diffFormatted` in `Journey`?

Comment (thevirtuoso1973) [reply app/src/main/java/com/jetbrains/handson/mpp/mobile/TrainDataAdapter.kt:98]: Not for now.

**Pull request merged**